### PR TITLE
Matrix Notification

### PR DESCRIPTION
**Pull request by antibios**
### Motivation for changes:
To receive notifications to a matrix server

### Detailed changes:
- Add additional notification plugin
- Add matrix-client python library

### Addressed issues:
- Fixes # .

### Implemented feature requests:
- Feathub #[XX](https://feathub.com/Flexget/Flexget/+XX).
- N/A

### Config usage if relevant (new plugin or updated schema):
```
      notify:
        entries:
          via:
            - matrix:
                server: "https://matrix.org"
                token: sender's token
                roomid: room identifier
```
### Log and/or tests output (preferably both):
```
paste output here
```
#### To Do:
Possibly remove dependency on matrix-client and just use http calls instead.
- [ ] Stuff..

### CODE REVIEW

File: flexget/components/notify/notifiers/matrix.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        try:
+            from matrix_client.api import MatrixHttpApi  # noqa
+        except ImportError as e:
+            logger.debug('Error importing MatrixHttpApi: {}', e)
+            raise DependencyError(
+                plugin_name, 'matrix_client', 'matrix_client module required. ImportError: %s' % e
+            )
+        #TODO: Is dns required to be loaded for the matrix_client module?
+        try:
+            import dns  # noqa
```
Comment (crawln45) [inline flexget/components/notify/notifiers/matrix.py:60]: Don't believe this import of DNS is needed. As it's never implemented into the call, and all DNS Queries would be handled by the MatrixHttpApi module and not by flexget.

**Review by antibios — COMMENTED** [review]
I've removed the references to external python modules and made a direct web call to matrix room.

File: flexget/components/notify/notifiers/matrix.py (hunk 1 above)
Comment (antibios) [reply flexget/components/notify/notifiers/matrix.py]: I've updated to remove the external deps

File: requirements.in [hunk 2]
```diff
@@ -31,3 +31,5 @@ zxcvbn-python
 progressbar==2.5
 more-itertools
 psutil>=5.8.0
+matrix_client>=0.3.2
```
[the remark this answers is not in the document]
Comment (antibios) [reply requirements.in]: I've updated to remove the external deps

**Pull request closed without merging**